pkg/downloader/chart_downloader.go

```diff
@@ -97,7 +97,32 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 
 	name := filepath.Base(u.Path)
 	destfile := filepath.Join(dest, name)
-	if err := ioutil.WriteFile(destfile, data.Bytes(), 0644); err != nil {
+	tmpfile, err := ioutil.TempFile(dest, "tmp")
+	// Remove the temp file in case an error occurred before renaming.
+	defer func() {
+		if tmpfile != nil {
+			if _, err := os.Stat(tmpfile.Name()); err == nil {
+				os.Remove(tmpfile.Name())
+			}
+		}
+	}()
+	if err != nil {
+		return "", nil, err
+	}
+
+	tmpfilename := tmpfile.Name()
+	if _, err := tmpfile.Write(data.Bytes()); err != nil {
```

Comment:
generally speaking the more idiomatic Go approach is to call `io.Copy(tmpfile, data)` instead. Can we make that change here?